cmdutil/metrics/otel/otel.go

```diff
@@ -23,24 +20,33 @@ func MustProvider(ctx context.Context, logger logrus.FieldLogger, cfg Config, se
 		logger.Fatal("provider collectorURL cannot be nil")
 	}
 
-	client := otel.NewHTTPClient(*cfg.CollectorURL)
-	expOpts := otlpmetric.WithMetricExportKindSelector(metric.DeltaExportKindSelector())
-	exporter := otlpmetric.NewUnstarted(client, expOpts)
-
-	attrs := []attribute.KeyValue{}
+	// configure some optional resource attributes
+	attrs := otel.MetricsDestinations(cfg.MetricsDestinations)
 	if cfg.Honeycomb.MetricsDataset != "" {
-		attrs = append(attrs, attribute.String("dataset", cfg.Honeycomb.MetricsDataset))
-	}
-	for _, md := range cfg.MetricsDestinations {
-		attrs = append(attrs, attribute.String(md, "true"))
+		attrs = append(attrs, otel.HoneycombDataset(cfg.Honeycomb.MetricsDataset))
 	}
 
 	allOpts := []otel.Option{
-		otel.WithExporter(exporter),
+		// ensure we have service.id, service.namespace, and service.instance.id attributes
+		otel.WithOpenTelemetryStandardService(service, serviceNamespace, serviceInstanceID),
+
+		// ensure we have _service and component attributes
+		otel.WithServiceStandard(service),
+
+		// ensure we have stage and _subcomponent attributes
+		otel.WithEnvironmentStandard(stage),
```

Comment:
Are we missing the subcomponent argument?

Reply:
I think `_subcomponent` is actually a typo from before. I updated this comment to `_subservice`. The option here sets both values, just like before.